ui/src/app/views/workspace-share/component.spec.ts

```diff
@@ -147,5 +147,11 @@ describe('WorkspaceShareComponent', () => {
         .toEqual(WorkspaceAccessLevel.OWNER);
   }));
 
-
+  it('validates and allows usernames or email addresses', fakeAsync(() => {
+    let asEmailAddress = workspaceSharePage.fixture.componentRef.instance.convertToEmail('bob');
```

Comment:
A better test would be to enter text and then click the share button, then assert a share call was made with the right email, rather than testing the private helper function.

The helper function is trivial enough that I don't think it justifies having a unit test.

Sorry - this may be a deviation from a lot of the unit tests we have currently but I think doing this correctly here will help improve future test code and our testing story overall.

Reply:
:+1: 
I will do that. I agree.